### PR TITLE
upgraded scala to 2.12 & added migrations

### LINKED ISSUE #3318: Upgrade Scala
https://scala-ci.typesafe.com/grafana/dashboard/db/scala-benchmark

Therefore also have to upgrade wkw: https://github.com/scalableminds/webknossos-wrap/pull/22

### PR DESCRIPTION
**Pull request by jstriebel**
~~Missing steps:
Release wkw (https://github.com/scalableminds/webknossos-wrap/pull/22), and adapt dependency version here (remove `-SNAPSHOT`).~~ done

This runs & compiles locally with a locally published wkw.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Test most critical parts. Compiler & CI should catch most stuff.

### Issues:
- fixes #3318

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- [x] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: app/oxalis/security/WkSilhouetteEnvironment.scala [hunk 1]
```diff
@@ -48,11 +46,20 @@ class WkSilhouetteEnvironment @Inject()(conf: WkConf,
   )
 
   val fingerprintGenerator = new DefaultFingerprintGenerator(false)
-  val idGenerator = new CompactRandomIDGenerator
-  val bearerTokenAuthenticatorDAO = new BearerTokenAuthenticatorRepository(tokenDAO)
+  val idGenerator = new CompactRandomIDGenerator()(executionContext)
```
Comment (fm3) [inline app/oxalis/security/WkSilhouetteEnvironment.scala]: Why do we need to pass the executionContext explicitly now?

Comment (fm3) [reply app/oxalis/security/WkSilhouetteEnvironment.scala]: I see, thanks. In this case I would prefer removing the 
`import scala.concurrent.ExecutionContext.Implicits.global`
(which I should have done already when adding the injected executionContext).
Could you do that in these instances?

Comment (jstriebel) [reply app/oxalis/security/WkSilhouetteEnvironment.scala]: Good idea, it worked here.

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/BucketProvider.scala [hunk 2]
```diff
@@ -17,7 +17,7 @@ import scala.concurrent.{Await, ExecutionContext, Future}
 
 trait BucketProvider extends FoxImplicits with LazyLogging {
 
-  def loadFromUnderlying(readInstruction: DataReadInstruction)(implicit ec: ExecutionContext): Fox[Cube] = Fox.empty
+  def loadFromUnderlying(readInstruction: DataReadInstruction)(implicit ec: ExecutionContext): Fox[Cube] = Fox.empty(ec)
```
Comment (jstriebel) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/BucketProvider.scala]: @fm3 Removing `Implicits.global` does not work for `BucketProvider.scala`, as it seems to be needed elsewhere.

**Pull request merged**